File: GreenSignal/Data/Models/Incident.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Reflection.PortableExecutable;

namespace Data.Models
{
    public class Incident
    {
        [Key]
        public Guid Id { get; set; }

        [MaxLength(10000)]
        public string Description { get; set; }

        [MaxLength(10000)]
        public string Address { get; set; }

        public double? Lat { get; set; }

        public double? Lng { get; set; }

        public Guid ReportedById { get; set; }
        public Citizen ReportedBy { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }

        [Required]
        public IncidentStatus Status { get; set; }

        [Required]
        public IncidentKind Kind { get; set; }

        //Дата взятия в работу
        public DateTime? BindingDate { get; set; }

        public Guid? InspectorId { get; set; }
        public Inspector? Inspector { get; set; }

        public IEnumerable<IncidentAttachment> IncidentAttachments { get; set; }
    }

    public enum IncidentKind
    {
        [Description("Загрязнение воздуха")]
        AirPollution,
        [Description("Загрязнение почвы")]
        SoilPollution,
        [Description("Добыча недр")]
        Excavation,
        [Description("Свавлка")]
        Dump,
        [Description("Вырубка лесов")]
        TreeCutting,
        [Description("Радиация")]
        Radiation
    }

    public enum IncidentStatus
    {
        [Description("Черновик")]
        Draft,
        [Description("Подтверждёна")]
        Submitted,
        [Description("Прикреплёна")]
        Attached,
        [Description("Завершена")]
        Completed,
        [Description("Закрыта")]
        Closed,
        [Description("Удалена")]
        Deleted
    }

    public enum ReportType
    {
        [Description("Нерешаемая")]
        Unsolvable,

        [Description("Нерешаемая")]
        Unacceptable
    }
}

File: GreenSignal.Mobile/app/src/main/java/com/example/greensignal/domain/model/response/Incident.kt
package com.example.greensignal.domain.model.response

import com.example.greensignal.data.remote.dto.response.IncidentAttachmentDto
import com.example.greensignal.data.remote.dto.response.IncidentDto
import com.example.greensignal.data.remote.dto.response.IncidentKind
import com.example.greensignal.data.remote.dto.response.IncidentReportStatisticDto
import com.example.greensignal.data.remote.dto.response.IncidentStatus
import java.time.LocalDateTime
import java.util.Date

data class Incident(
    val id: String = "",
    val description: String = "",
    val address: String = "",
    val lat: Double? = null,
    val lng: Double? = null,
    val reportedById: String = "",
    val createdAt: Date = Date(),
    val status: IncidentStatus = IncidentStatus.Submitted,
    val kind: IncidentKind = IncidentKind.Dump,
    val bindingDate: Date? = null,
    val inspectorId: String? = null,
    val incidentAttachments: MutableList<IncidentAttachment> = mutableListOf(),
    val reportedBy: Citizen = Citizen()

)

fun IncidentDto.toIncident(): Incident {
    return  Incident(
        id = id,
        description = description,
        address = address,
        lat = lat,
        lng = lng,
        reportedById = reportedById,
        createdAt = createdAt,
        status = status,
        kind = kind,
        bindingDate = bindingDate,
        inspectorId = inspectorId,
        incidentAttachments = incidentAttachments.map { x -> x.toIncidentAttachment() }.toMutableList(),
        reportedBy = reportedBy.toCitizen()
    )
}